main.cpp

```diff
@@ -92,6 +92,76 @@ good to go!
  */
 
 #include <iostream>
+
+struct FloatType
+{
+    float add( float lhs, float rhs )
+    {
+        return lhs + rhs;
+    }
+    float subtract( float lhs, float rhs )
+    {
+        return lhs - rhs;
+    }
+    float multiply( float lhs, float rhs )
+    {
+        return lhs * rhs;
+    }
+    float divide( float lhs, float rhs )
+    {   
+        if (rhs == 0.0f) std::cout << "\nwarning, floating point division by zero returns 'inf' !" << std::endl;
+        return lhs / rhs;
+    }
+};
+
+struct DoubleType
+{
+    double add( double lhs, double rhs )
+    {
+        return lhs + rhs;
+    }
+    double subtract( double lhs, double rhs )
+    {
+        return lhs - rhs;
+    }
+    double multiply( double lhs, double rhs )
+    {
+        return lhs * rhs;
+    }
+    double divide( double lhs, double rhs )
+    {   
+        if (rhs == 0.0) std::cout << "\nwarning, floating point division by zero returns 'inf' !" << std::endl;
```

Comment:
put `if()` bodies on their own line to improve the readability of the code
- [ ] revise